.github/shared/src/doc-preview.js

```diff
@@ -44,12 +44,12 @@ const SPEC_FILE_REGEX =
 /**
  * Extract swagger file metadata from path.
  * @param {string} specPath
- * @returns {SwaggerFileMetadata}
+ * @returns {SwaggerFileMetadata | null}
  */
 export function parseSwaggerFilePath(specPath) {
   const m = specPath.match(SPEC_FILE_REGEX);
   if (!m) {
-    throw new Error(`Path "${specPath}" does not match expected swagger file pattern.`);
+    return null;
```

Comment:
how about we keep this throwing, but callers can wrap in try/catch if they want to ignore errors?